### PR TITLE
[runtime-security] do not rely on ticker to handle remaining events in queue

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This PR changes the way we handled event remaining in the queue. For such events we only use the ticker to insert placeholder event so that real within the WindowSize will be handled.

The previous time based algorithm led to handle to much event due to time drift between kernel and user space.

### Motivation

We have seen lot of event order issue in staging environment.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Metric has been added when an event is not handled in the right order.

### CODE REVIEW

File: pkg/security/probe/reorderer.go [hunk 1]
```diff
@@ -187,14 +191,16 @@ func (r *ReOrderer) HandleEvent(CPU int, data []byte, perfMap *manager.PerfMap,
...
 	return &ReOrderer{
-		queue:            make(chan []byte, opts.QueueSize),
-		handler:          handler,
-		list:             &reOrdererList{},
-		pool:             &reOrdererNodePool{},
-		timestampGetter:  tsg,
-		resolveTimestamp: rts,
-		opts:             opts,
+		queue:   make(chan []byte, opts.QueueSize),
+		handler: handler,
```
Comment (Gui774ume) [inline pkg/security/probe/reorderer.go:197]: Should we pass the handler to the `reOrdererList` directly ?

File: pkg/security/probe/probe_bpf.go [hunk 2]
```diff
@@ -295,6 +297,12 @@ func (p *Probe) handleEvent(data []byte) {
 	}
 	offset += read
 
+	// check event order
+	if event.TimestampRaw < p.lastTimestamp && p.lastTimestamp != 0 {
+		_ = p.statsdClient.Count(MetricPrefix+".event.reorder_error", 1, []string{}, 1.0)
```
Comment (Gui774ume) [inline pkg/security/probe/probe_bpf.go:302]: Nit: to follow the pattern of the other metrics, you might want to call it `.events.`. I would also call it `differed` or `sorting_error` just to make sure we know what kind of error this is ... but that's really just a nit :)

**Review by Gui774ume — APPROVED** [review]
lgtm :+1:

File: pkg/security/probe/probe_bpf.go (hunk 2 above)
Comment (lebauce) [reply pkg/security/probe/probe_bpf.go:302]: +1 for the s/error/errors/

**Pull request merged**